scvi/external/scbasset/_model.py

```diff
@@ -192,6 +203,193 @@ def get_cell_bias(self) -> np.ndarray:
 
+    @classmethod
+    def _download_motifs(cls, genome: str, motif_dir: str) -> None:
+        """Download a set of motifs injected into peaks"""
+        logger.info(f"Downloading motif set to: {motif_dir}")
+        # download the motif set
+        url_name = cls.MOTIF_URLS.get(genome, None)  # (url, dir_name)
+        if url_name is None:
+            raise ValueError(f"{genome} is not a supported motif set.")
+        urlretrieve(url_name[0], filename=Path(motif_dir, f"{genome}_motifs.tar.gz"))
+        # untar it
+        import tarfile
+
+        def rename_members(tarball):
+            """Rename files in the tarball to remove the top level folder"""
+            for member in tarball.getmembers():
+                if member.path.startswith(url_name[1]):
+                    member.path = member.path.replace(url_name[1] + "/", "")
+                    yield member
+
+        # importing the "tarfile" module
+        tarball = tarfile.open(Path(motif_dir, f"{genome}_motifs.tar.gz"))
+        tarball.extractall(path=motif_dir, members=rename_members(tarball))
+        tarball.close()
+
+        # `motif_dir` now has `shuffled_peaks_motifs` as a subdir and
+        # `shuffled_peaks.fasta` as a root level file.
+        logger.info("Download and extraction complete.")
+        return
+
+    def _get_motif_library(
+        self, tf: str, genome: str = "human", motif_dir: str = None
+    ) -> Tuple[List[str], List[str]]:
+        """Load sequences with a TF motif injected from a pre-computed library
+
+        Parameters
+        ----------
+        tf : str
+            name of the transcription factor motif to load. Must be present in a
+            pre-computed library.
+        genome : str, optional
+            species name for the motif injection procedure. Currently, only "human"
+            is supported.
+        motif_dir : str, optional
+            path for the motif library. Will download if not already present.
+
+        Returns
+        -------
+        motif_seqs : List[str]
+            list of sequences with an injected motif.
+        bg_seqs : List[str]
+            dinucleotide shuffled background sequences.
+        """
+        if motif_dir is None:
+            motif_dir = self.DEFAULT_MOTIF_DIR
+
+        # ensure input is a `Path` object
+        motif_dir = Path(motif_dir)
+        if not Path(motif_dir, "shuffled_peaks.fasta").exists():
+            motif_dir.mkdir(exist_ok=True, parents=True)
+            self._download_motifs(genome=genome, motif_dir=motif_dir)
+
+        fasta_files = motif_dir.glob("shuffled_peaks_motifs/*.fasta")
+        tf_names = [f.stem for f in fasta_files]
+        if tf not in tf_names:
+            msg = f"{tf} is not found as a motif in the library."
+            raise ValueError(msg)
+
+        # load the motifs
+        tf_motif_path = Path(motif_dir, "shuffled_peaks_motifs", f"{tf}.fasta")
+        motif_seqs = list(SeqIO.parse(tf_motif_path, "fasta"))
```

Comment:
might be best to import biopython things in this function

we have a dependencies decorator here

https://github.com/jacobkimmel/scvi/blob/a3ac84016ad2f30e481edd944c1cf65e2f00a4cd/scvi/_decorators.py#L18-L19

This can be used like `@dependencies("biopython")`

Reply:
also very cool, added